src/shared/util.js

```diff
@@ -1218,20 +1218,164 @@ function MessageHandler(sourceName, targetName, comObj) {
   this.sourceName = sourceName;
   this.targetName = targetName;
   this.comObj = comObj;
-  this.callbackIndex = 1;
+  this.callbackId = 1;
+  this.streamId = 1;
   this.postMessageTransfers = true;
-  var callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
-  var ah = this.actionHandler = Object.create(null);
+  let streamSinks = this.streamSinks = Object.create(null);
+  let streamControllers = this.streamControllers = Object.create(null);
+  let callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
+  let ah = this.actionHandler = Object.create(null);
+
+  let _processStreamMessage = (data) => {
```

Comment:
> I might be totally misunderstanding, but I'm interpreting #8430 (comment) as suggesting that they be moved into MessageHandler.prototype instead (since the updated patch doesn't really change anything otherwise)!?

I think i misunderstood [#8439 (comment)](https://github.com/mozilla/pdf.js/pull/8430#discussion_r119618166), sorry for that. I will move `_processStreamMessage` and `_createStreamSink` to `MessageHandler.prototype`